packages/taco/src/conditions/base/rpc.ts

```diff
@@ -3,20 +3,24 @@ import { z } from 'zod';
 import { SUPPORTED_CHAIN_IDS } from '../const';
 import createUnionSchema from '../zod';
 
-import { EthAddressOrUserAddressSchema, returnValueTestSchema } from './shared';
+import {
+  EthAddressOrUserAddressSchema,
+  paramOrContextParamSchema,
+  returnValueTestSchema,
+} from './shared';
 
 export const RpcConditionType = 'rpc';
 
 export const rpcConditionSchema = z.object({
   conditionType: z.literal(RpcConditionType).default(RpcConditionType),
   chain: createUnionSchema(SUPPORTED_CHAIN_IDS),
-  method: z.enum(['eth_getBalance', 'balanceOf']),
+  method: z.enum(['eth_getBalance']),
   parameters: z.union([
-    z.array(EthAddressOrUserAddressSchema).length(1),
+    z.array(EthAddressOrUserAddressSchema).nonempty(),
     // Using tuple here because ordering matters
-    z.tuple([EthAddressOrUserAddressSchema, z.union([z.string(), z.number()])]),
+    z.tuple([EthAddressOrUserAddressSchema, paramOrContextParamSchema]),
   ]),
-  returnValueTest: returnValueTestSchema,
+  returnValueTest: returnValueTestSchema, // Update to allow multiple return values after expanding supported methods
```

Comment:
Hmmm no, it will only be updated once we update the `method` field in the RPC schema

Reply:
I thought the `returnValueTestSchema` is from here - https://github.com/piotr-roslaniec/nucypher-ts/blob/fix-lynx/packages/taco/src/conditions/base/shared.ts#L37? In which case tuples of values etc. are allowed (same as parameters). Or am I mistaken?

Reply:
Ah no, you're right. It is covered, but I was hoping we could constrain it instead of relaxing it (as we did in the latest change) when we have more information about the available methods.

Reply:
You can, I guess, if you have rpcConditionSchema define it's own returnValueSchema...?